Rename Worktree class to WorkingTree

jcouball opened the pull request:
### Description

Rename the `Worktree` class to `WorkingTree` to be more consistent with Git documentation and the term's use in public.

See [the definition of Working Tree in the Git glossary](https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefworkingtreeaworkingtree) or [search the web for Git working tree](http://www.google.com/search?q=git+working+tree).

### What is changed?
All code and documentation that referred to worktree was changed to working tree.

### What else was changed and why?
* Updated the description of the RubyGit module in the project's README.md and in lib/ruby_git.rb
* Added the SVG version of the RubyGit Class Diagram so people don't have to look at it in LucidChart

jcouball committed:
Rename Worktree class to WorkingTree

frankthrock commented on lib/ruby_git.rb:
```diff
@@ -55,78 +55,79 @@ class << self
   end
 
-  # Create an empty Git repository under the root worktree `path`
+  # Create an empty Git repository under the root woring tree `path`
```
Spelling error.

frankthrock commented on README.md:
```diff
@@ -8,9 +8,10 @@
-RubyGit is an object-oriented wrapper for the `git` command line tool for working with Worktrees
-and Repositories. It tries to make more sense out of the Git command line. See the object model
-in [this Lucid chart diagram](https://app.lucidchart.com/invitations/accept/7df13bab-3383-4683-8cb4-e76d539de93d)
+RubyGit is an object-oriented wrapper for the `git` command line tool for working git Repositories.
```
Perhaps you want to add 'with' here?  ...working WITH git repositories.  (Up to you...)

jcouball commented:
Thanks, @frankthrock for the review.  I have fixed the issues you pointed out and added the following:

* Updated the description of the RubyGit module in the README.md
* Updated the description of the RubyGit module in the documentation in lib/ruby_git.rb
* Added the SVG for the RubyGit class diagram so people don't have to view it in Lucid Charts.  I updated the link in the README.md to point to this SVG instead of the Lucid Charts version.

frankthrock reviewed (approved):
Looks really good.  I like the verbose changes.
